catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java

```diff
@@ -873,4 +882,21 @@ public String getSortKey(String field) {
   public Stream<String> anyTextFields() {
     return anyTextFieldsCache.stream();
   }
+
+  /**
+   * Gets the metadata size limit from custom.system.properties. Defaults to 5mb limit.
+   *
+   * @return Integer metadata size limit in bytes
+   */
+  private Integer getMetadataSizeLimit() {
+    String sizeLimit = System.getProperty("metadata.size.limit");
```

Comment:
❗️ `System.getProperty` is a synchronized call.  You do not want to call this every time `addFields` is called.  This value should be read and cached in the constructor or a static block.

Reply:
Definitely do not want to do that. fixed this to make one call in a static final constant.